attribute.js

```diff
@@ -0,0 +1,59 @@
+import { isEqual } from 'lodash';
+
+class BaseAttribute {
+  constructor({ parent, value }) {
+    this._parent = parent;
+
+    constructorValues.call(this, value);
+    constructorTriggers.call(this);
+  }
+
+  getValue(value) { return value; }
+
+  parse(value) { return value; }
+
+  reset() {
+    this.set(this.getOriginalValue());
+    this.setPristine();
+  }
+
+  setPristine() {
+    this.setOriginalValuePristine();
+    this.isDirty = false;
+    this.hasChanged = false;
+  }
+
+  setValue(newValue, oldValue) {
+    if (!isEqual(oldValue, newValue)) {
+      this.hasChanged = !isEqual(this.getOriginalValue(), newValue);
+      this.isDirty = true;
+      this._parent.isDirty = true;
+    }
+
+    return this.parse(newValue);
```

Comment:
missing the assignation into the right attribute

Reply:
setValue is just an exposed method used by the constructor for the `set` method (line 49)